virttest/qemu_vm.py

```diff
@@ -1252,6 +1252,13 @@ def set_value(opt_string, key, fallback=None):
 
             return secret_cmdline + " -spice %s" % (",".join(spice_opts))
 
+        def add_thread_context(devices, params):
+            devs = []
+            dev = devices.thread_context_define_by_params(params, "tc1")
```

Comment:
Here the ID has been hardcoded, how should I proceed? Thanks!

Reply:
To make it configurable usually we would introduce a param to let user specify the id of the logical device(s), see the example below. And such a param is also in charge of pointing out the counts of devices, usually, from 0 up to N.

Looks we could introduce a new one for the thread context objects, and I'd suggest using `vm_thread_contexts` or `vm_thread_context_objects` as the name.

(btw, can we add multiple thread context objects to a guest?)

https://github.com/avocado-framework/avocado-vt/blob/fb39a11288331854cfdbfc757a407f0274c56322/virttest/qemu_vm.py#L2233

```
throttle_groups = tg1 tg2 tg3
```

Reply:
Updated now, thanks!

Reply:
> (btw, can we add multiple thread context objects to a guest?)

@luckyh actually it's supported by QEMU, but let me confirm if it is feasible/makes sense